app/controllers/templates_controller.rb

```diff
@@ -79,15 +79,21 @@ def auto_complete_controller_name
   end
 
   def preview
-    base = @template.preview_host_collection
-    @host = params[:preview_host_id].present? ? base.find(params[:preview_host_id]) : base.first
+    if params[:id]
+      find_resource
+      base = @template.preview_host_collection
+      @host = params[:preview_host_id].present? ? base.find(params[:preview_host_id]) : base.first
+    else
+      type = request.original_url.split("/")[4]
+      @host = Host.find(params[:preview_host_id])
+      @template = resource_class.new(params[type])
```

Comment:
Is type needed here for anything? Foreman only renders `ProvisioningTemplate`.  Partition tables (`Ptable`) usually don't contain any ERB so there is no need to render them. 

@ares could we even move this `preview` method entirely to `ProvisioningTemplatesController`?